### DISCUSSION TITLE
DE analysis on multiple genotype/treatment conditions

### DISCUSSION THREAD

**Discussion by lhoranportelance** [satijalab/seurat#7703] [Q&A]
Hi everyone, this is my first time using Seurat for a big sc-RNAseq analysis, and so far, the experience has been seamless! However, I've run into a bit of an issue when it comes to now doing differential expression analysis. I have performed everything through clustering and assigning names to clusters, and I am now looking to do DE analysis between genotype/treatment groups within various clusters. I have checked the various vignettes linked in other posts similar to this but nothing there seems to address my specific scenario. 

I have 4 conditions: WT_vehicle, WT_treated, Transgenic_vehicle, Transgenic_treated. I have assigned these with labels for genotype (WT or transgenic) and treatment (vehicle or treated), as well as a combined label of "group", which combines the genotype and treatment into one label ("WT_vehicle"). I would now like to do DE analysis on these different groups, within the same cluster--for example, Transgenic_vehicle vs. Transgenic_treated in cluster 3. 

I figure I need to use FindMarkers(), but I'm not sure what to use as idents.1 and idents.2, and how to specify the cluster in which I want the DE analysis to occur. 

Any help you can provide on this is much appreciated :) can't thank everyone enough for posting on here and being such a great working community!

**Comment by wudustan** [comment]
You can set up a 1vsOther analysis fairly simply using `FindAllMarkers()`. This will test one group against a background of the others. It will pull the group assignments from `Idents()` so you can set your group assignments like that.

Alternatively you could set up some kind of pairwise comparison between all groups:

    group.names <- c("WT_vehicle", "WT_treated", "Transgenic_vehicle", "Transgenic_treated")
    group.pairs <- sort(combn(group.names, 2, FUN = paste, collapse = "-"))

    pairs.group.res <- lapply(group.pairs, function(x){
  
      message("Doing ", x)
  
      FindMarkers(object = seurat.object,
                  ident.1 = strsplit(x, split = "-")[[1]][1],
                  ident.2 = strsplit(x, split = "-")[[1]][2],
                  ...)
    })

There's probably a much simpler way to feed the pairs into lapply. Let me know if you find something more elegant.
